### PR TITLE
DBZ-7786 Add MariaDB connector documentation

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browses/DBZ-7786

@roldanbob, as we discussed, I duplicated the MySQL documentation and commented on the parts that were shared.  The one part I did not comment on was the properties. I think it's reasonable for us to compare the properties between the two connectors. The shared ones can be moved to a shared include file, and those that do not remain specific to the appropriate connector.  Hopefully, that's okay.

### CODE REVIEW

**Comment by roldanbob** [comment]
@Naros Quick update so that you don't think this fell into a black hole. 
I started work on this, but after talking to others in the docs group, I learned about similar work that was done on another project. So, for now, rather than reinvent the wheel, I'm putting pausing work, pending a meeting early next week to discuss an alternate approach. If the other approach doesn't dovetail well with the structure of the upstream repo, I'll resume work on the current track.

**Comment by roldanbob** [comment]
@Naros I've tested using shared content in the MariaDB docs, and it's looking good -- at least in the upstream.  Next step is to validate that things work in the downstream build. If I'm able to build successfully, I'll propagate the changes to MySQL. 
One thing that I'm a little bit unhappy about is that after implementing the properties as description lists, it's a little dissonant to go from looking at the Required and Advanced properties that are in that format to viewing the remaining properties (schema history, Kafka signals, sink notifications, and so forth), which are still formatted as tables. Not ideal in terms of consistency. But as a POC, I think the current work is a good incremental step, and it provides a roadmap for how we might move those other properties sets to shared lists.

**Comment by roldanbob** [comment]
The latest changes address ID collisions and linking problems that caused the downstream build to fail. The MariaDB connector doc now uses shared content, including shared lists of required and advanced configuration properties, and builds upstream and downstream. 
The next step is to replace content in the MySQL docs with the shared content.

**Comment by roldanbob** [comment]
I *think* we're good on this. I updated the MySQL doc so that it shares content the same as the MariaDB docs, and tested it in upstream and downstream builds. The downstream build threw some errors at first, but I refreshed the branch and just got a clean build. 🤞🏼 Let's let this marinate until I return on the 8th, and if I can reproduce that success, we'll be ready to merge this. 
Note that I did make a few content changes along the way. Nothing that requires major technical review, but if you have a chance to give it a once over, that might be good. Thanks.

**Comment by roldanbob** [comment]
Setting up the two connectors to share content was pretty straightforward for the community version, but turned out to be way more complex for the downstream. I hadn't anticipated that xrefs hosted in the shared content are not parsed if the link target has a ModuleID defined for it. As a result, I've removed ModuleID declarations in the source for several topics.

**Comment by roldanbob** [comment]
Tested in local Antora and downstream builds.

This should be ready to merge now.

File: documentation/modules/ROOT/partials/modules/all-connectors/shared-mariadb-mysql.adoc [hunk 1]
```diff
@@ -0,0 +1,3137 @@
...
+
+You can choose to produce events for a subset of the schemas and tables in a database.
+Optionally, you can ignore, mask, or truncate columns that contain sensitive data, that are larger than a specified size, or that you do not need.
+
+[source,json]
+----
+{
+    "name": "inventory-connector", // <1>
+    "config": {
+        "connector.class": "io.debezium.connector.{context}.MariaDbConnector", // <2>
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/shared-mariadb-mysql.adoc:2930]:
```suggestion
        "connector.class": "io.debezium.connector.{context}.{connector-name}Connector", // <2>
```

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-incremental-snapshot-metrics.adoc [hunk 2]
```diff
@@ -25,3 +25,8 @@ The connector also provides the following additional snapshot metrics when an in
 |The upper bound of the primary key set of the currently snapshotted table.
 
 |===
+
+ifdef::MARIADB,MYSQL[]
+The {prodname} MariaDB connector also provides the `HoldingGlobalLock` custom snapshot metric.
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-incremental-snapshot-metrics.adoc:30]:
```suggestion
The {prodname} {connector-name} connector also provides the `HoldingGlobalLock` custom snapshot metric.
```

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc [hunk 3]
```diff
@@ -75,3 +75,50 @@ The values will incorporate any differences between the clocks on the machines w
...
+|`string`
+|The name of the binlog file that the connector has most recently read.
+
+|[[binlog-position]]<<binlog-position,`+BinlogPosition+`>>
+|`long`
+|The most recent position (in bytes) within the binlog that the connector has read.
+
+|[[is-gtid-mode-enabled]]<<is-gtid-mode-enabled,`+IsGtidModeEnabled+`>>
+|`boolean`
+|Flag that denotes whether the connector is currently tracking GTIDs from MariaDB server.
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc:97]:
```suggestion
|Flag that denotes whether the connector is currently tracking GTIDs from {connector-name} server.
```

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc [hunk 4]
```diff
@@ -75,3 +75,50 @@ The values will incorporate any differences between the clocks on the machines w
...
+
+|[[gtid-set]]<<gtid-set,`+GtidSet+`>>
+|`string`
+|The string representation of the most recent GTID set processed by the connector when reading the binlog.
+
+|[[number-of-skipped-events]]<<number-of-skipped-events,`+NumberOfSkippedEvents+`>>
+|`long`
+|The number of events that have been skipped by the MariaDB connector. Typically events are skipped due to a malformed or unparseable event from MariaDB's binlog.
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc:105]:
```suggestion
|The number of events that have been skipped by the {connector-name} connector. Typically events are skipped due to a malformed or unparseable event from {connector-name}'s binlog.
```

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc [hunk 5]
```diff
@@ -75,3 +75,50 @@ The values will incorporate any differences between the clocks on the machines w
...
+
+|[[number-of-disconnects]]<<number-of-disconnects,`+NumberOfDisconnects+`>>
+|`long`
+|The number of disconnects by the MariaDB connector.
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc:109]:
```suggestion
|The number of disconnects by the {connector-name} connector.
```

**Review by Naros — COMMENTED** [review]
Hi @roldanbob, I found a few places where the connector-name was not being set properly, otherwise LGTM.

**Comment by roldanbob** [comment]
> Hi @roldanbob, I found a few places where the connector-name was not being set properly, otherwise LGTM.

Oh, thanks for catching that. I'm definitely a bit bleary-eyed after testing this pretty much non-stop for the past week or so. Sorry for the multiple commits, but the option to batch them didn't seem to be available.

**Comment by Naros** [comment]
Applied to main (=3.0) and branch 2.7.

**Pull request merged**